### PR TITLE
#20 Profile page & more

### PR DESCRIPTION
**Pull request by mbullington**
Started as profile page but got somewhat carried away.

- Implement profile page and reauth.
- Upgrade all dependencies. Is it possible to use yarn in the future? I recommend completely removing node_modules then running `yarn install` to get latest dependencies.
- On the new version segmented control wasn't working so I replaced it.
- Some style tweaks I thought of over the past like 4 months of working on this on-and-off.
- Made the logo bigger.

This is a really big PR, so if you have time, please test nearly everything.

### CODE REVIEW

File: src/components/LoginGuard.tsx [hunk 1]
```diff
@@ -1,15 +1,50 @@
...
+) {
+  if (!validateEmail(email)) {
+    return;
+  }
+
+  if (operation == SIGN_IN) {
+    try {
+      const user = await AuthService.signIn(email, password);
+      if (!user) {
+        console.log("Test2");
```
[the remark this answers is not in the document]
Comment (mbullington) [reply src/components/LoginGuard.tsx:28]: Probably I'll take a look

[the remark this answers is not in the document]
Comment (rsquared226) [reply src/components/LoginGuard.tsx:28]: Oh I meant only the console.log, not the other stuff

**Comment by mbullington** [comment]
I switched it to use https://github.com/react-native-community/segmented-control .

Personally, I don't think it looks as nice, but at least it's stable. @rsquared226 

Screenshot_20200915-112706
Screen Shot 2020-09-15 at 11 26 59 AM

**Review by rsquared226 — APPROVED** [review]
Everything works! (lol ignore warnings)
The only thing is that this warning pops up in the map page now which I feel like just be because of some dependency update. We don't really need the map page though, should we just comment that out in another PR? (maybe a task for Ali or Jon to get sorta familiar with the code?)

image
image

**Comment by mbullington** [comment]
I agree, sounds good.

**Pull request merged**